src/CPUSolver.cpp

```diff
@@ -748,6 +748,11 @@ void CPUSolver::tallyScalarFlux(segment* curr_segment, int azim_index,
   FP_PRECISION* sigma_t = curr_segment->_material->getSigmaT();
   FP_PRECISION delta_psi, exponential;
 
+  /* Extract weights for all polar angles */
+  FP_PRECISION weights[_num_polar];
```

Comment:
If I inline the ``Quadrature::getWeight(...)`` function I would probably need to get rid of the error checking conditionals. Do you think I should do that?